idb/postgres/postgres_integration_test.go

```diff
@@ -1502,35 +1519,66 @@ func TestAddBlockAppOptInOutSameRound(t *testing.T) {
 }
 
 // TestSearchForInnerTransactionReturnsRootTransaction checks that the parent
-// transaction is returned when matching on inner transactions.
+// transaction is returned when matching on inner transactions if the
+// ReturnInnerTxnFlag is false. If the ReturnInnerTxnFlag is true, it should
+// return the inner txn instead.
 func TestSearchForInnerTransactionReturnsRootTransaction(t *testing.T) {
 	var appAddr basics.Address
 	appAddr[1] = 99
 
 	tests := []struct {
-		name    string
-		matches int
-		filter  idb.TransactionFilter
+		name        string
+		matches     int
+		returnInner bool
```

Comment:
I just added a flag to the existing tests here and checked if the inner txn was being returned.

Reply:
nice :+1: